### DISCUSSION TITLE
Increasing / decreasing circle size over time mod proposal

### DISCUSSION THREAD

**Discussion by vruss** [ppy/osu#13108] [Ideas]
**Describe the new feature:** 
A mod that changes your CS over time. Similar to the wind-up and wind-down mod.

**Proposal designs of the feature:** 
It could start of very small and gradually increase in size. I think this could be a _fun_ challenge for players. I also think such a mod could be combined with other mods.
The names for such a mod could be Grow or Shrink but they are already taken.

Not relevant for this proposal but if possible, maybe the CS could reflect the intensity of the song?

**Comment by vruss** [comment]
Is it possible to use the existing [ModTimeRamp](https://github.com/ppy/osu/blob/master/osu.Game/Rulesets/Mods/ModTimeRamp.cs) class currently used by wind-up mod to scale the CS as the map progresses? Or is a new ramp class needed?

**Comment by peppy** [comment]
The circle size would need to be applied differently to each hitobject, so a new implementation is required.

**Comment by bdach** [comment]
There's been of influx of requests for changing beatmap parameters and gameplay elements in different ways. Care needs to be taken in deciding whether something should be a mod and *how* it should be presented to the user, so that the mod collection does not become an inscrutable mess.

**Comment by recognitionmark** [comment]
The problem is, if one would decrease it to 0 and play a dense map (e.g. The Big Black) it would be VERY difficult.

**Comment by vruss** [comment]
You could limit it within reasonable limits. And it's not for ranked gameplay.
